### PR TITLE
chore: add product carousel

### LINKED ISSUE #575: [Carousel] Implement Product Carousel Block
Used on almost all category page, like https://franklin.moleculardevices.com/products/microplate-readers/washers

### PR DESCRIPTION
**Pull request by mhaack**
Adds product carousel (without compare) used on product categories and events pages.

Fix #575
Relates to #523

Test URL with large list of products
- Before: https://main--moleculardevices--hlxsites.hlx.page/products/microplate-readers
- After: https://product-carousel--moleculardevices--hlxsites.hlx.page/products/microplate-readers

Test URL with short list of products
- Before: https://main--moleculardevices--hlxsites.hlx.page/products/microplate-readers/washers
- After: https://product-carousel--moleculardevices--hlxsites.hlx.page/products/microplate-readers/washers

Test URL with short list of products
- Before: https://main--moleculardevices--hlxsites.hlx.page/events/biophysical-society-baltimore
- After: https://product-carousel--moleculardevices--hlxsites.hlx.page/events/biophysical-society-baltimore

### CODE REVIEW

**Comment by jose-correia** [comment]
In the first example https://product-carousel--moleculardevices--hlxsites.hlx.page/products/microplate-readers are we ditching the original behaviour where we can go from the first item to the last by clicking the left arrow?
I remember we already supported this in the card carousel code

File: blocks/product-carousel/product-carousel.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  });
+
+  await createCarousel(
+    block,
+    products,
+    {
+      defaultStyling: true,
+      navButtons: window.matchMedia('only screen and (max-width: 767px)').matches || products.length > 3,
+      dotButtons: false,
+      infiniteScroll: false,
```
Comment (jose-correia) [inline blocks/product-carousel/product-carousel.js:26]: this should be true no?

**Review by jose-correia — CHANGES_REQUESTED** [review]

File: blocks/product-carousel/product-carousel.js (hunk 1 above)
Comment (mhaack) [reply blocks/product-carousel/product-carousel.js]: You are right, fixed

**Review by andreituicu — APPROVED** [review]
There are some defects, please see the comments below. 

Overall the code looks ok.

File: blocks/product-carousel/product-carousel.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    defaultButtonText: 'Details',
+    c2aLinkStyle: true,
+  });
+
+  await createCarousel(
+    block,
+    products,
+    {
+      defaultStyling: true,
+      navButtons: window.matchMedia('only screen and (max-width: 767px)').matches || products.length > 3,
```
[the remark this answers is not in the document]
Comment (mhaack) [reply blocks/product-carousel/product-carousel.js:24]: Fixed in combination with `infiniteScroll: products.length > 3` for <= 3 there is no scroll on desktop and items should not repeat. Only on smaller screens scroll is needed.

**Review by jose-correia — APPROVED** [review]

**Pull request merged**